lib/functions.php

```diff
@@ -309,6 +309,10 @@ function ($matches) {
         $uri
     );
 
+    if (null === $uri) {
+        throw new InvalidUriException('Invalid, or could not parse URI');
```

Comment:
Note: this keeps phpstan happy, otherwise it gets worried that `preg_replace_callback` might return `null` and that would cause trouble in later code that uses `$uri`. I couldn't find a value for the input parameter `$uri` that will actually cause `null` - I have to pass in a string (otherwise PHP will get an error on the call itself). So this seems untestable.